### PR TITLE
Feature/legal notice

### PR DESCRIPTION
**Pull request by Cardamced**
## Description

Please include a summary of the change or which issue is fixed.

## Type of change

- [X] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Area of change

- [X] Frontend
- [  ] Backend
- [X] Other: .....

## General checklist:

- [X] My code follows the style guidelines of this project
- [X] I ran `bun run format`/`bun run lint` before commit
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation where needed
- [X] My changes generate no new warnings
- [X] I have checked my code and corrected any misspellings
- [X] I have updated `develop` and merged to my branch before submitting pull request
- [X] My pull request generate no conflicts with `develop` branch
- [X] I requested code review from other team members

## Frontend checklist:

- [X] I followed guidelines for `HTML`/`JSX`/`Typescript`
- [X] My `Javascript` generate no new console errors
- [X] I tested my code cross browsers
- [ ] My slice is pixel perfect for both desktop and mobile according to design
- [ ] I conducted basic QA to assure all features are working
- [ ] I tested responsive for mobile and tablet resolutions

## Backend checklist:

- [ ] I tested admin by manually adding content from zero
- [ ] I created easy to use admin experience which is self-explanatory
- [ ] I added description to admin fields in hard-to-understand areas
- [ ] I followed guidelines for naming `Typescript` variables
- [ ] I conducted basic QA to assure all features are working

### CODE REVIEW

**Review by jaalazard — DISMISSED** [review]
Nice ! Et responsive en plus !

File: packages/frontend/web/src/pages/LegalModal.tsx
Comment (Justicesvr) [inline packages/frontend/web/src/pages/LegalModal.tsx:1]: bien vu le local storage

**Review by Justicesvr — DISMISSED** [review]

File: packages/frontend/web/src/pages/LegalModal.tsx [hunk 1]
```diff
@@ -0,0 +1,180 @@
+import { useEffect, useState } from 'react';
+
+export default function LegalNotice() {
+  const [isModalShown, setIsModalShown] = useState(true);
+
+  // checks LocalStorage when component is mounted.
+  // It can store simple datas like user's preferences
+  // and would be cleared as long as the user shuts down his browser.
+  useEffect(() => {
+    const hideModal: string | null = localStorage.getItem('hideLegalNotice');
```
Comment (YDrogen) [inline packages/frontend/web/src/pages/LegalModal.tsx]: le kebab-case convient bien pour les noms dans le localstorage

File: packages/frontend/web/src/pages/LegalModal.tsx [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+    }
+  }, []);
+
+  const handleClose = () => {
+    // registers user's choice into browser's LocalStorage
+    localStorage.setItem('hideLegalNotice', 'true');
+    setIsModalShown(false);
+  };
+
+  return isModalShown ? (
```
Comment (YDrogen) [inline packages/frontend/web/src/pages/LegalModal.tsx]: un if est plus clair qu'une ternaire

File: packages/frontend/web/src/pages/LegalModal.tsx [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+    <div className=' flex flex-col items-center text-black'>
```
Comment (YDrogen) [inline packages/frontend/web/src/pages/LegalModal.tsx]: espaces ?

File: packages/frontend/web/src/pages/LegalModal.tsx [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+      <div className='z-50 flex h-screen w-screen flex-col items-center justify-center gap-6 text-4xl'>
+        <div className=' bg-light-green xxs:rounded-xl xxs:text-xl m-2 rounded-[30px] border-[5px] border-black p-4 font-bold transition-transform duration-500 ease-in-out hover:rotate-1 hover:scale-110 hover:justify-normal hover:bg-opacity-80 sm:rounded-3xl sm:text-6xl'>
```
Comment (YDrogen) [inline packages/frontend/web/src/pages/LegalModal.tsx]: espaces ?

**Review by YDrogen — CHANGES_REQUESTED** [review]

File: packages/frontend/web/src/pages/LegalModal.tsx (hunk 1 above)
Comment (Cardamced) [reply packages/frontend/web/src/pages/LegalModal.tsx]: C'est corrigé.

File: packages/frontend/web/src/pages/LegalModal.tsx (hunk 2 above)
Comment (Cardamced) [reply packages/frontend/web/src/pages/LegalModal.tsx]: C'est corrigé.

File: packages/frontend/web/src/pages/LegalModal.tsx (hunk 3 above)
Comment (Cardamced) [reply packages/frontend/web/src/pages/LegalModal.tsx]: Yep, c'est enlevé.

File: packages/frontend/web/src/pages/LegalModal.tsx (hunk 4 above)
Comment (Cardamced) [reply packages/frontend/web/src/pages/LegalModal.tsx]: Corrections done.

**Review by jaalazard — DISMISSED** [review]
mouais

**Review by Justicesvr — DISMISSED** [review]

File: packages/frontend/web/src/pages/LegalModal.tsx [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+    }
+  }, []);
+
+  const handleClose = () => {
+    // registers user's choice into browser's LocalStorage
+    localStorage.setItem('hide-legal-notice', 'true');
+    setIsModalShown(false);
+  };
+
+  if (isModalShown) {
```
Comment (YDrogen) [inline packages/frontend/web/src/pages/LegalModal.tsx]: L'idée du `if` ici est de faire un return préventif
"si modal ne s'affiche pas, return null"
Ça évite de devoir encapsuler tout le JSX dans un if

**Review by YDrogen — CHANGES_REQUESTED** [review]

File: packages/frontend/web/src/pages/LegalModal.tsx (hunk 5 above)
Comment (Cardamced) [reply packages/frontend/web/src/pages/LegalModal.tsx]: Hinhin, je note.

**Review by jaalazard — APPROVED** [review]

**Review by Justicesvr — APPROVED** [review]

**Review by YDrogen — APPROVED** [review]

**Pull request merged**